storage/postgres/postgres.go

```diff
@@ -236,8 +236,9 @@ func (p *Postgres) ReadUserTuple(ctx context.Context, store string, tupleKey *op
 	defer span.End()
 
 	objectType, objectID := tupleUtils.SplitObject(tupleKey.GetObject())
-	row := p.pool.QueryRow(ctx, `SELECT object_type, object_id, relation, _user FROM tuple WHERE store = $1 AND object_type = $2 AND object_id = $3 AND relation = $4 AND _user = $5 AND user_type = $6`,
-		store, objectType, objectID, tupleKey.GetRelation(), tupleKey.GetUser(), tupleUtils.User)
+	row := p.pool.QueryRow(ctx, `SELECT object_type, object_id, relation, _user FROM tuple WHERE store = $1 AND object_type = $2 AND object_id = $3 AND relation = $4 AND _user = $5`,
```

Comment:
Usertype should still be provided but extracted from the tupleKey. This way we guarantee we land in the correct Index https://github.com/openfga/openfga/blob/b7d987f12a5e8f093a7eefa52bfddea349c9f966/assets/migrations/postgres/001_initialize_schema.sql#L14-L16

Reply:
Good call. I've updated it to use the `user_type` column as a filter. Let me know how that looks.